sdk/communication/azure-communication-sms/src/test/java/com/azure/communication/sms/SmsAsyncClientTests.java

```diff
@@ -1,90 +1,180 @@
+
 package com.azure.communication.sms;
 
-import com.azure.communication.common.PhoneNumberIdentifier;
-import com.azure.communication.sms.models.SendSmsOptions;
+
+import com.azure.communication.sms.models.SmsSendOptions;
+import com.azure.core.credential.TokenCredential;
+import com.azure.identity.DefaultAzureCredentialBuilder;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import com.azure.core.http.HttpClient;
+import reactor.core.publisher.Mono;
+import reactor.test.StepVerifier;
 
 import java.util.ArrayList;
 import java.util.List;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
 
-import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Test;
+public class SmsAsyncClientTests extends SmsTestBase {
+    private List<String> to;
+    private SmsAsyncClient asyncClient;
 
-import reactor.test.StepVerifier;
+    @Override
+    protected void beforeTest() {
+        super.beforeTest();
+    }
 
-public class SmsAsyncClientTests extends SmsTestBase {
+    @ParameterizedTest
+    @MethodSource("com.azure.core.test.TestBase#getHttpClients")
+    public void createAsyncClientUsingConnectionString(HttpClient httpClient) {
+
+        to = new ArrayList<String>();
+        to.add(TO_PHONE_NUMBER);
+        SmsClientBuilder builder = getSmsClientUsingConnectionString(httpClient);
+        asyncClient = setupAsyncClient(builder, "createAsyncSmsClientUsingConnectionString");
+        assertNotNull(asyncClient);
+        Mono<Iterable<SmsSendResult>> response = asyncClient.send(FROM_PHONE_NUMBER, to, MESSAGE, null);
+        StepVerifier.create(response)
+            .assertNext(item -> {
+                assertNotNull(item);
+            })
+            .verifyComplete();
+
+
+    }
+
+    @ParameterizedTest
+    @MethodSource("com.azure.core.test.TestBase#getHttpClients")
+    public void sendSmsUsingTokenCredential(HttpClient httpClient) {
+        TokenCredential tokenCredential = new DefaultAzureCredentialBuilder().build();
+        to = new ArrayList<>();
+        to.add(TO_PHONE_NUMBER);
+        SmsClientBuilder  builder = getSmsClientWithToken(httpClient, tokenCredential);
+        asyncClient = setupAsyncClient(builder, "createAsyncSmsClientUsingConnectionString");
+        assertNotNull(asyncClient);
+        Mono<Iterable<SmsSendResult>> response = asyncClient.send(FROM_PHONE_NUMBER, to, MESSAGE, null);
+        StepVerifier.create(response)
+            .assertNext(item -> {
+                assertNotNull(item);
+            })
+            .verifyComplete();
+
+
+    }
+    @ParameterizedTest
+    @MethodSource("com.azure.core.test.TestBase#getHttpClients")
+    public void sendToMultipleUsers(HttpClient httpClient) {
+
+        to = new ArrayList<String>();
+        to.add(TO_PHONE_NUMBER);
+        // Arrange
+        SmsClientBuilder builder = getSmsClient(httpClient);
+        asyncClient = setupAsyncClient(builder, "send");
```

Comment:
The names that you pass in should be the name of the test, and it should be unique for all the test cases